modelon/impact/client/entities/experiment.py

```diff
@@ -100,6 +100,16 @@ class _Workflow(enum.Enum):
     CLASS_BASED = "CLASS_BASED"
 
 
+class ExperimentSizeInfo:
+    def __init__(self, data: Dict[str, Any]):
```

Comment:
I'd suggest parsing out the integer directly and raising exception if the expected key is not present. It feels unnecessary to delay that until the "total" call and have Optional[int]

Reply:
I was thinking a bit about extensibility. If the "sizeInfo" object contains more data in the future... But I agree on throwing an exception is a better approach here. I have also made ExperimentSizeInfo to a dataclass now.